### ISSUE TITLE
[$500] [MEDIUM] Workspace - When adding a member to WS fails, a "Chat Report" conversation is created

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#34056]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.22-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Open New Expensify app
2. Log in with HT account @applause.expensifail.com
3. Create WS
4. Add some nonexistent @applause.expensifail.com members to the created WS
5. Remove the error of adding members

## Expected Result:
If an error occurs in adding a participant to WS, the created conversation must be deleted from LHN or archived, preserving its original form

## Actual Result:
When removing the error of adding a member to WS, the "Chat Report" conversation remains

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/0c8999c9-4e8e-4529-9193-56eff5434d68

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ff2d94edb1b3acf6</li>
        <li>Upwork Job ID: 1743364368062525440</li>
        <li>Last Price Increase: 2024-01-19</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @AndrewGable</details>

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
If we add a WS member and it fails, closing/clearing the error doesn't remove the expense chat.

### What is the root cause of that problem?
When we press the clear/close error message, we only remove them from POLICY_MEMBERS and PERSONAL_DETAILS_LIST onyx
https://github.com/Expensify/App/blob/f5a3c0202128a8e0e77607defbbdcb4f30f80ad3/src/libs/actions/Policy.js#L997-L1004

but we never clear the optimistic expense chat that we add when adding the member.

### What changes do you think we should make in order to solve the problem?

Remove the optimistic expense chat in `clearAddMemberError`.

**Comment by kadiealexander** [comment]
@jjcoffee bump to review these proposals please!

**Comment by jjcoffee** [comment]
Reviewing tomorrow!

**Comment by jjcoffee** [comment]
I'm leaning towards the alternative solution in @bernhardoj's [proposal](https://github.com/Expensify/App/issues/34056#issuecomment-1879209187), as it sounds like it would be a nice straightforward solution. 

I am a bit unsure on the exact expected result here, however. It looks like the report does actually get created on the BE, at least for the emails I tested. Can you confirm @bernhardoj?

**Comment by bernhardoj** [comment]
> It looks like the report does actually get created on the BE, at least for the emails I tested.

I tested by inviting a VoIP phone number and after relogin, the policy expense chat is not available anymore, indicating it's removed from the BE.

**Comment by jjcoffee** [comment]
Hmm what error do you get when inviting that number? For an "incorrect" email (e.g. `applausetester+100124@applause.expensifail.com` and insert the current time at the end of the address part) I get `404 Account not found`. I'm not really sure what triggers this error, maybe the assigned engineer can clarify.

When logging out and back in and opening the Workspace members list I also get a 404 error and an endless loader (even if I previously dismissed the initial error on adding the member).

The workspace-member chat also persists, and shows up in search, so I suspect we may need some BE changes here too to actually remove the report for this case. That is if this is actually a case we need to handle!

I've _also_ tried an invalid number `+84566123459@expensify.sms` which gives this error:

![image](https://github.com/Expensify/App/assets/27287420/8dec6a24-21aa-4772-823b-5883a6f0bfc2)

This behaves more as you describe, with the chat disappearing on relogin (and it's not accessible via deeplink).

@tienifr's [proposal](https://github.com/Expensify/App/issues/34056#issuecomment-1879560840) is correct though in that for multiple members the whole `AddMembersToWorkspace` request fails (on current main) and all members show as having an error. I think your alternative solution won't handle this @bernhardoj.

**Comment by bernhardoj** [comment]
> Hmm what error do you get when inviting that number?

<img width="442" alt="image" src="https://github.com/Expensify/App/assets/50919443/15e10cf0-bb92-46fc-9e04-668d7533c501">


> When logging out and back in and opening the Workspace members list I also get a 404 error and an endless loader (even if I previously dismissed the initial error on adding the member).

I see. I can reproduce this with the email.

> multiple members the whole AddMembersToWorkspace request fails (on current main) and all members show as having an error

If the request fails, all members won't be added to the workspace.

**Comment by jjcoffee** [comment]
> When logging out and back in and opening the Workspace members list I also get a 404 error and an endless loader (even if I previously dismissed the initial error on adding the member).

Confirmed that this only happens for emails (`+84566123459@expensify.sms` works as expected). Despite the error, and the fact that we hide the member if the error is dismissed, they do actually get added to the workspace on the BE (they also appear in oldDot). I think that needs to be handled with a BE fix.

> If the request fails, all members won't be added to the workspace (at least I don't see them after refreshing/reopening the page).

Hmm so it looks like it actually depends on the order in which you invite people, so if the first invite is to a "real" email (that succeeds), followed by any number of fake/bad numbers, then the real email _will_ get added (appears on oldDot, and the report persists after relogin). Another BE fix required there!

In summary, I'm wondering if it might be best to go with a fully BE fix here where the BE returns the relevant Onyx data to remove the optimistically created reports (since there are a couple BE changes required anyway). 

If we want a faster fix I'm good with going with the alt. solution in @bernhardoj's [proposal](https://github.com/Expensify/App/issues/34056#issuecomment-1879209187) which doesn't require any BE changes to work, but with the caveat that it will remove optimistically created expense chats for all invited members if the request fails (even if some of the members did get added). It's not perfect, but it's better than before and those chats would re-appear after relogin!

:ribbon::eyes::ribbon: C+ reviewed

**Comment by kadiealexander** [comment]
@AndrewGable bump!

**Comment by kadiealexander** [comment]
Not overdue. @jjcoffee @greg-schroeder bump!

**Comment by kadiealexander** [comment]
Not overdue. @jjcoffee @greg-schroeder bump!

**Comment by jjcoffee** [comment]
@tienifr I do agree that there is also an issue where _all_ members have an error message added, which is worth fixing and which your proposal deals with. However, I'm not sure if it fits the scope of this issue, which deals with created chats not being cleared. In any case we would deal with that after the BE changes are made.

I think that we should probably treat these as two separate issues, though in order to avoid overcomplicating this.  Curious if @AndrewGable or @greg-schroeder have thoughts on that?

**Comment by kadiealexander** [comment]
@greg-schroeder @AndrewGable any updates on the question above? I'm inclined to agree with @tienifr.

**Comment by greg-schroeder** [comment]
I'm making this a Hot Pick for an internal engineer to grab

**Comment by joelbettner** [comment]
@jjcoffee I'm not able to reproduce this locally, and I'm not sure what I'm doing differently.

My goal is to pinpoint where, in the back end, we create the chat report, and prevent that from happening if the user wasn't added to the workspace.

However, every time I try to add a user to a policy, the user is getting added successfully. Do I have to be using a @applause.expensifail.com email address for some specific reason?

Also, the video in the OP no longer works. Is there any chance we can get another video of the error happening?

**Comment by joelbettner** [comment]
Ok...I did some extensive digging through the logs on the date that I think the issue was made. I found this...


```
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:17 920 | applausetester+050101su@applause.expensifail.com | Processing 'AddMembersToWorkspace' for 'expensify.com' from '46.34.192.144' ~~ command: 'AddMembersToWorkspace' employees: '[{"email":"applausetester+070122su@applause.expensifail.com"},{"email":"applausetester+070123su@applause.expensifail.com"},{"email":"applausetester+070121su@applause.expensifail.com"},{"email":"applausetester+070120su@applause.expensifail.com"},{"email":"applausetester+070124su@applause.expensifail.com"}]' welcomeNote: 'You have been invited to Applause's Workspace! Download the Expensify mobile app at use.expensify.com/download to start tracking your expenses.' policyID: '11B7033054EA2A83' reportCreationData: '{"applausetester+070122su@applause.expensifail.com":{"reportID":"8333008110649611","reportActionID":"5301199207452452868"},"applausetester+070123su@applause.expensifail.com":{"reportID":"5778842199005229","reportActionID":"2847203940668099504"},"applausetester+070121su@applause.expensifail.com":{"reportID":"2873950575063","reportActionID":"896972681456206227"},"applausetester+070120su@applause.expensifail.com":{"reportID":"4682810349347657","reportActionID":"4605029967145525787"},"applausetester+070124su@applause.expensifail.com":{"reportID":"3815259058569139","reportActionID":"2226380527874968089"}}' appversion: '1.4.22-0' pusherSocketID: '635883.325873' authToken: '<REDACTED>' referer: 'ecash' platform: 'desktop' api_setCookie: 'false' email: 'applausetester+050101su@applause.expensifail.com' partnerName: 'expensify.com' browserGUID: '659854cde0c22' HTTP_CF_BOT_SCORE: '94' HTTP_CF_JA3_HASH: 'b269dc0d2c27af73eb6c4adc13ae243f'
...
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:18 000 | applausetester+050101su@applause.expensifail.com | Bedrock\Client - Starting a request ~~ command: 'CreateAccount' clusterName: 'auth' headers: '[authToken: '<REDACTED>' partnerName: 'expensify.com' partnerPassword: '<REDACTED>' sso: '<REDACTED>' email: 'applausetester+070122su@applause.expensifail.com' referer: '(share policy)' password: '<REDACTED>' securityGroupID: '' personalPolicyValue: '<REDACTED>' isOnWalletBeta: '1' template: 'NO_EMAIL' ignoreUnvalidatedSecondary: '' isOriginatingFromECash: '' publicPersonalDetails: '' shouldSetClassicRedirectNVP: '' createConciergeChatReport: '' logParam: 'applausetester+050101su@applause.expensifail.com' urlToNewDot: 'https://staging.new.expensify.com/' maxNumberOfUpdates: '500' commitCount: '21940982438' requestID: '840e09a60c007a69-DME' lastIP: '46.34.192.144' writeConsistency: 'ASYNC' priority: '500' timeout: '110000']'
...
840e09a60c007a69-DME | db1.sjc | 2024-01-05 19:13:18 152 | applausetester+050101su@applause.expensifail.com | Waiting for 'CreateAccount' to complete.
...
840e09a60c007a69-DME | db1.sjc | 2024-01-05 19:13:18 153 | applausetester+050101su@applause.expensifail.com | Domain controlled account being created in a closed state. domainAccountID=5857801
...
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 441 | applausetester+050101su@applause.expensifail.com | Bedrock\Client - Starting a request ~~ command: 'GetRestrictedAuthToken' clusterName: 'auth' headers: '[email: 'applausetester+070122su@applause.expensifail.com' type: 'receiptUpload' validateCode: '<REDACTED>' partnerName: 'expensify.com' idempotent: '1' logParam: 'applausetester+050101su@applause.expensifail.com' urlToNewDot: 'https://staging.new.expensify.com/' maxNumberOfUpdates: '500' commitCount: '21940982953' requestID: '840e09a60c007a69-DME' lastIP: '46.34.192.144' writeConsistency: 'ASYNC' priority: '500' timeout: '110000']'
...
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 444 | applausetester+050101su@applause.expensifail.com | Auth - Call GetRestrictedAuthToken return 404
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 444 | applausetester+050101su@applause.expensifail.com | Throw ExpException - 5a298b2fb32635da318ee93f6e1feeee ~~ message: '404 Account not found' exceptionMessage: 'Auth GetRestrictedAuthTokenByEmail returned an error' exceptionFile: '/git/releases/expensify.com/3cf1c64/lib/Auth.php' exceptionLine: '125' exceptionCode: '404'
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 444 | applausetester+050101su@applause.expensifail.com | [trace] StackTrace start
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 444 | applausetester+050101su@applause.expensifail.com | [trace] 0/4. file '/git/releases/expensify.com/3cf1c64/lib/Auth.php' function '_standardResponseHandler' line '2508'
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 444 | applausetester+050101su@applause.expensifail.com | [trace] 1/4. file '/git/releases/expensify.com/3cf1c64/lib/AuthTokenManager.php' function 'getRestrictedAuthTokenByEmail' line '555'
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 444 | applausetester+050101su@applause.expensifail.com | [trace] 2/4. file '/git/releases/expensify.com/3cf1c64/lib/AuthTokenManager.php' function 'getRestrictedAuthTokenByEmail' line '241'
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 444 | applausetester+050101su@applause.expensifail.com | [trace] 3/4. file '/git/releases/expensify.com/3cf1c64/lib/PolicyAPI.php' function 'getEmailToken' line '1329'
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 444 | applausetester+050101su@applause.expensifail.com | [trace] 4/4. file '/git/releases/expensify.com/3cf1c64/api.php' function 'addMembersToWorkspace' line '2051'
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 444 | applausetester+050101su@applause.expensifail.com | [trace] StackTrace end
840e09a60c007a69-DME | staging-www1.lax | 2024-01-05 19:13:20 444 | applausetester+050101su@applause.expensifail.com | [OOPS!] API Threw: Auth GetRestrictedAuthTokenByEmail returned an error ExpException - 5a298b2fb32635da318ee93f6e1feeee ~~ message: '404 Account not found' exceptionMessage: 'Auth GetRestrictedAuthTokenByEmail returned an error' exceptionFile: '/git/releases/expensify.com/3cf1c64/lib/Auth.php' exceptionLine: '125' exceptionCode: '404'
```

I think the key here is this line:
```
Domain controlled account being created in a closed state. domainAccountID=5857801
```

I'll keep digging.

**Comment by joelbettner** [comment]
I confirmed that we created an entry in the accounts table, but not in the logins table:
```sql
joel@db2.rno:~$ sudo readdb.sh "SELECT * FROM accounts WHERE email = 'applausetester+070122su@applause.expensifail.com';"
created              accountID  email                                             password  ssn  dob  promoAmount  validated  validateCode                                                                                       closed               lastLogin            subscribed  inviterID  twoFactorAuthValidated  twoFactorAuthSecretKey  twoFactorAuthRecoveryCodes
-------------------  ---------  ------------------------------------------------  --------  ---  ---  -----------  ---------  -------------------------------------------------------------------------------------------------  -------------------  -------------------  ----------  ---------  ----------------------  ----------------------  --------------------------
2024-01-05 19:13:18  16210889   applausetester+070122su@applause.expensifail.com                      0            0          $argon2id$v=19$m=19456,t=2,p=1$+1A1ko2c3vRCcAlC+B4pkQ$gCktLoz/LHLwN+aMhwg+OASFHd/PGIt5/V2W2NH4M8w  2024-01-05 19:13:18  2024-01-05 19:43:18  1           16207559   0                                                                         
joel@db2.rno:~$
joel@db2.rno:~$ sudo readdb.sh "SELECT * FROM logins WHERE accountID = 16210889;"
joel@db2.rno:~$
```

So, this is making sense. We created the account correctly, but left it "closed", then we created a report for it, then we failed when we tried to get the restricted auth token because it returned that there was no user with that accountID [here](https://github.com/Expensify/Auth/blob/2d7ff973067fa5d1055aa28ad35078345f4c3e86/auth/command/GetRestrictedAuthToken.cpp#L135).

**Comment by joelbettner** [comment]
I believe I found a solution for this, and I don't think it requires any front end changes. With this solution, we should no longer get the same error that we got before when adding a user that does not yet exist to the workspace (because that user gets created anyway).

**Comment by luacmartins** [comment]
Taking over from Joel

**Comment by greg-schroeder** [comment]
Cleaning up Wave 6 issues (there are hundreds) - adding users to a Workspace is an admin-only action, so setting this to Wave 8. Will let you decide on priority @zanyrenney!

**Comment by Christinadobrzyn** [comment]
Hi there - can we merge [this issue](https://github.com/Expensify/App/issues/35538) into this job? 

It seems like both GHs involve the 'Auth GetRestrictedAuthTokenByEmail returned an error' error.

**Comment by kadiealexander** [comment]
@jjcoffee could you please check the issue linked above and see if it would have the same fix?

**Comment by jjcoffee** [comment]
That issue doesn't appear to be related, the error isn't the same and it's an entirely different flow. cc @Christinadobrzyn

**Comment by jjcoffee** [comment]
https://github.com/Expensify/App/issues/25592 seems to be related, I think we can either hold on that or ask Applause for a retest cc @kadiealexander

**Comment by kadiealexander** [comment]
Have asked Applause to retest!

**Comment by luacmartins** [comment]
@kavimuru could you please retest this issue?

**Comment by kavimuru** [comment]
Not reproducible now

https://github.com/Expensify/App/assets/43996225/c12d4de2-a019-4afe-861e-c860ae5a136e

**Comment by luacmartins** [comment]
Cool, seems like we can close this issue then. Thank you!
